cob_cartesian_controller/src/trajectory_profile_generator/trajectory_profile_generator.cpp

```diff
@@ -0,0 +1,472 @@
+            ROS_INFO_STREAM("Se = 0");
+
+            array.push_back(pa.getStartValue());
+        }
+    }
+    pa.setArray(array);
+    return true;
+}
+
+inline bool TrajectoryProfileRamp::calculateProfile(std::vector<double> path_matrix[4],
+                                                    const double Se, const double Se_roll, const double Se_pitch, const double Se_yaw,
+                                                    geometry_msgs::Pose start)
+{
+    double accl = params_.profile.accl;
+    double vel  = params_.profile.vel;
+    unsigned int max_steps;
+
+    //Convert to RPY
+    double roll_start, pitch_start, yaw_start;
+    tf::Quaternion q;
+    tf::quaternionMsgToTF(start.orientation, q);
+    tf::Matrix3x3(q).getRPY(roll_start, pitch_start, yaw_start);
+
+    std::vector<double> linear_path, roll_path, pitch_path, yaw_path;
+
+    cob_cartesian_controller::PathArray lin(0, Se, 0, linear_path);
+    cob_cartesian_controller::PathArray roll(1, Se_roll, roll_start, roll_path);
+    cob_cartesian_controller::PathArray pitch(2, Se_pitch, pitch_start, pitch_path);
+    cob_cartesian_controller::PathArray yaw(3, Se_yaw, yaw_start, yaw_path);
+    cob_cartesian_controller::PathMatrix pm(lin,roll,pitch,yaw);
+
+    std::vector<cob_cartesian_controller::PathArray> sortedMatrix = pm.getSortedMatrix();   // Sort the Matrix from the largest Se (0) to the smallest one (3)
+
+    ROS_INFO("Got sorted_matrix.");
+
+    this->getProfileTimings(std::fabs(sortedMatrix[0].getSe()), accl, vel);  // Calculate the velocity profile timings with respect to the largest Se
+
+    ROS_INFO("Got profil timings.");
+
+    // Calculate the paths
+    for(int i=0; i<sortedMatrix.size(); i++)
+    {
+        this->generatePath(sortedMatrix[i]);
+    }
+
+    ROS_INFO("GeneratePath success.");
+
+    // Resize the path vectors
+    for(int i = 0; i < sortedMatrix.size(); i++)
+    {
+        max_steps = std::max((unsigned int)sortedMatrix[i].getArray().size() , max_steps);
+    }
+
+    ROS_INFO("max_steps success.");
+
+    // Re-adjust the Matrix to its originally form. Index 0 to 3
+    std::vector<double> temp_array;
+    cob_cartesian_controller::PathArray temp(0, 0, 0.0, temp_array);
+
+    for(int j = 0; j < sortedMatrix.size(); j++)
+    {
+        for(int i = 0; i < sortedMatrix.size()-1; i++)
+        {
+            if(sortedMatrix[i].getIdx() > sortedMatrix[i+1].getIdx())
+            {
+                temp = sortedMatrix[i];
+                sortedMatrix[i] = sortedMatrix[i+1];
+                sortedMatrix[i+1] = temp;
+            }
+        }
+    }
+
+    ROS_INFO("IndexSort success.");
+
+    for(int i = 0; i < sortedMatrix.size(); i++)
+    {
+        path_matrix[i] = sortedMatrix[i].getArray();
+    }
+
+    ROS_INFO("Fill path_matrix success.");
+
+    for(int i = 0; i < sortedMatrix.size(); i++)
+    {
+        if(path_matrix[i].size() < max_steps)
+        {
+            path_matrix[i].resize(max_steps, path_matrix[i].back());
```

Comment:
So here you "fix" the array lengths....
In what case is this step required? I mean as all the Se- and ProfileTimings computation has been done...shouldn't that  guarantee enough that all PathArrays have same length?
If they don't, there must be something wrong....